projects/doc-lib/src/source-code-viewer/source-code-viewer.component.ts

```diff
@@ -3,48 +3,57 @@
  * SPDX-License-Identifier: BSD-2-Clause
  */
 
-import { Component, Input, Type } from '@angular/core';
-import { DocumentationRetrieverService } from '../documentation-retriever.service';
-import { HighlightService } from '../highlight/highlight.service';
+import {Component, ElementRef, Input, OnDestroy, Type} from '@angular/core';
+import {DocumentationRetrieverService} from '../documentation-retriever.service';
+import {TabbedCodeViewer} from '../vanilla/tabbed-code-viewer/tabbed-code-viewer.component';
+import Prism from 'prismjs';
+import 'prismjs/components/prism-typescript';
+import 'prismjs/components/prism-scss';
 
 /**
  * To display the 3 code parts(TypeScript, HTML and CSS) of a component
  */
 @Component({
     selector: 'vcd-source-code-viewer',
-    templateUrl: './source-code-viewer.component.html',
-    styleUrls: ['./source-code-viewer.component.scss'],
+    template: '',
 })
-export class SourceCodeViewerComponent {
-    /**
-     * Different source codes of the component
-     */
-    typescriptSourceCode: string;
-    htmlSourceCode: string;
-    cssSourceCode: string;
+export class SourceCodeViewerComponent implements OnDestroy {
+    constructor(private documentationRetriever: DocumentationRetrieverService, private el: ElementRef) {}
 
-    constructor(
-        private documentationRetriever: DocumentationRetrieverService,
-        private highlightService: HighlightService
-    ) {}
+    private codes: TabbedCodeViewer;
 
     /**
-     * The component which source (typescript, html, css) will be displayed
+     * The component whose typescript, html, css will be displayed
      */
     @Input()
     set component(component: Type<any>) {
         if (!component) {
             return;
         }
-        // TODO: externalize string literals
-        this.typescriptSourceCode = this.highlightService.highlightTypescript(
-            this.documentationRetriever.getTypescriptSourceCode(component)
-        );
-        this.htmlSourceCode = this.highlightService.highlightHtml(
-            this.documentationRetriever.getHtmlSourceCode(component) || 'No html found'
-        );
-        this.cssSourceCode = this.highlightService.highlightScss(
-            this.documentationRetriever.getCssSourceCode(component) || 'No CSS found'
+
+        this.codes = new TabbedCodeViewer(
+            [
+                {
+                    language: 'HTML',
+                    sourceCode: this.documentationRetriever.getHtmlSourceCode(component),
+                    languageId: Prism.languages.html,
```

Comment:
Right now we show Typescript as the default panel, this will default to HTML (?). Do we want that behavior?

Reply:
Yes, mentioned in the description 😄 HTML is the entry point even if most of the code is in the TypeScript